dict[] actions            # From Recipe
bool executed             # From RecipeHistory
int32 execution_order     # From RecipeHistory
bool new_actions
